### PR TITLE
JP Remote Install: Differentiate between bad username and bad password

### PR DESCRIPTION
**Pull request by seear**
**Before**
screen shot 2018-04-18 at 12 44 20

**After**
screen shot 2018-04-18 at 12 42 31screen shot 2018-04-18 at 12 42 51

Server-side patch D11976-code differentiates between different kinds of login failures. Take advantage of this to highlight the appropriate entry field. Also, for any login failure that is not a credentials problem, fall back to the unknown error message.

## Testing
* Sandbox the API and apply patch D11976-code
* Go to https://calypso.live/jetpack/connect?branch=update/jp-remote-install-creds-errors
* Enter URL of a dotorg site without Jetpack active
* On next screen, enter some combinations of bad password and username

### CODE REVIEW

File: client/jetpack-connect/connection-notice-types.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const NOT_EXISTS = 'notExists';
 export const NOT_JETPACK = 'notJetpack';
 export const NOT_WORDPRESS = 'notWordPress';
 export const OUTDATED_JETPACK = 'outdatedJetpack';
+export const INVALID_CREDENTIALS = 'invalidCredentials';
```
Comment (AnnaMag) [inline client/jetpack-connect/connection-notice-types.js]: ... which seems to be making `export const LOGIN_FAILURE = 'credsInvalid';` obsolete, am I right?

Comment (AnnaMag) [inline client/jetpack-connect/connection-notice-types.js]: The definition of `LOGIN_FAILURE` becomes obsolete, am I right?

Comment (seear) [reply client/jetpack-connect/connection-notice-types.js]: Oh well spotted. Seems like it should be `'loginFailure'` to match the others.

File: client/jetpack-connect/remote-credentials.js [hunk 2]
```diff
@@ -168,8 +172,8 @@ export class OrgCredentialsForm extends Component {
 		) {
 			return ACTIVATION_FAILURE;
 		}
-		if ( installError === 'LOGIN_FAILURE' ) {
```
Comment (AnnaMag) [inline client/jetpack-connect/remote-credentials.js]: I'd keep it as a comment above the UNKNOWN error case, for error coverage reference.

**Comment by joanrho** [comment]
@seear Would it be possible to add a red error text below the highlighted field saying "Username does not exist. Please try again." (for username error) and "Incorrect password. Please try again." (for password error)?

**Comment by seear** [comment]
> Would it be possible to add a red error text below the highlighted field saying "Username does not exist. Please try again." (for username error) and "Incorrect password. Please try again." (for password error)?

Sure. Added to #24279.

**Pull request merged**